### ISSUE TITLE
Using 'let' with rspec resulting in 'NoMethodError'

### ISSUE THREAD

**Issue by Nachtigall** [rspec/rspec-core#2054]
Hi *,

today, we have updated rspec-core from 3.0.3 to 3.3.2 version in our project and got problems during spec runs:
```ruby
 Failure/Error: Unable to find matching line from backtrace
     NoMethodError:
       undefined method `fetch_or_store' for nil:NilClass
```

As far i see, issue was already reported here (https://github.com/rspec/rspec-core/issues/1921), but looks like it was not fixed.

Thanks

**Comment by JonRowe** [comment]
As discussed in #1921 this is caused when using something that extends RSpec but uses a private part of our api, in the case of #1921 it was `rspec-retry`

**Comment by myronmarston** [comment]
#1921 was fixed via #1950 before 3.3 was released. In fact, no versions of RSpec have been released with that specific bug (although it appears you have hit a similar bug).

Can you put yogether a code example that demonstrates the issue? Without that there's no way for us to guess what causes this error for you but no one else.
